Exoft.Gamification.Api.Services/Services/AuthService.cs

```diff
@@ -0,0 +1,67 @@
+﻿using AutoMapper;
+using Exoft.Gamification.Api.Common.Helpers;
+using Exoft.Gamification.Api.Common.Models;
+using Exoft.Gamification.Api.Data.Core.Entities;
+using Exoft.Gamification.Api.Services.Interfaces;
+using Exoft.Gamification.Api.Services.Interfaces.Interfaces;
+using Exoft.Gamification.Api.Services.Services;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+
+namespace Exoft.Gamification.Api.Services
+{
+    public class AuthService : IAuthService
+    {
+        private readonly IUserService _userService;
+        private readonly IJwtSecret _jwtSecret;
+        private readonly IMapper mapper;
+        
+        public AuthService(IUserService userService, IJwtSecret jwtSecret, IMapper mapper)
```

Comment:
As far as UserService should model, it would be better to inject IUserRepository here